### PR TITLE
Reflect changes in Tuya's IoT Platform names and structure

### PR DESCRIPTION
**Pull request by drthanwho**
## Proposed change

1. Tuya IoT Platform has changed some names and structure of their pages that contradict with current instructions
    Updated names based on current edition of their platform. 
2. Addition of ΝΟΤΕ about being prompted to subscribe to a plan to clarify this is free. 
3. Fix of a rogue NOTE 😅
4. New screenshots to reflect the changes

📌 Needs to be added to 2021.10 milestone

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in current documentation (`current` branch).
- [ ] Updated documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/tuya.markdown [hunk 1]
```diff
@@ -48,19 +48,20 @@ You have created an account for [Tuya Smart or Smart Life app](https://developer
 
 **1.** Log in to the [Tuya IoT Platform](https://iot.tuya.com/).
 
-**2.** In the left navigation bar, click **Cloud** > **Projects**.
+**2.** In the left navigation bar, click **Cloud** > **Development**.
+<p class='note'>If this is your first time setting up a an account you will be asked to subscribe to a Cloud Development Plan. You can select the Trial Edition and "purchase" it (this is free). </p>
```
Comment (ludeeus) [inline source/_integrations/tuya.markdown]:
```suggestion
**2.** In the left navigation bar, click **Cloud** > **Development**.

<p class='note'>
If this is your first time setting up an account you will be asked to subscribe to a Cloud Development Plan. You can select the Trial Edition.
 </p>
```

File: source/_integrations/tuya.markdown [hunk 2]
```diff
@@ -48,19 +48,20 @@ You have created an account for [Tuya Smart or Smart Life app](https://developer
 
 **1.** Log in to the [Tuya IoT Platform](https://iot.tuya.com/).
```
Comment (ludeeus) [inline source/_integrations/tuya.markdown:49]: Add a note here that this portal requires a separate account from your app.

File: source/_integrations/tuya.markdown [hunk 3]
```diff
@@ -48,19 +48,20 @@ You have created an account for [Tuya Smart or Smart Life app](https://developer
...
 
-**2.** In the left navigation bar, click **Cloud** > **Projects**.
+**2.** In the left navigation bar, click **Cloud** > **Development**.
+<p class='note'>If this is your first time setting up a an account you will be asked to subscribe to a Cloud Development Plan. You can select the Trial Edition and "purchase" it (this is free). </p>
 
-**3.** On the page that appears, click **Create**.
+**3.** On the page that appears, click **Create Cloud Project**.
 
-**4.** In the **Create Project** dialog box, configure **Project Name**, **Description**, **Industry**, and **Availability Zone**. In the **Development Method** field, select **Smart Home PaaS** from the dropdown list. Select all available zones.
+**4.** In the **Create Project** dialog box, configure **Project Name**, **Description**, **Industry**, and **Data Center**. In the **Development Method** field, select **Smart Home** from the dropdown list. In the **Data Center** field, select all available zones.
```
Comment (ludeeus) [inline source/_integrations/tuya.markdown]:
```suggestion
**4.** In the **Create Project** dialog box, configure **Project Name**, **Description**, **Industry**, and **Data Center**. In the **Development Method** field, select **Smart Home** from the dropdown list. In the **Data Center** field, select the zone you are located in.
```

File: source/_integrations/tuya.markdown [hunk 4]
```diff
@@ -48,19 +48,20 @@ You have created an account for [Tuya Smart or Smart Life app](https://developer
...
 
 ![](/images/integrations/tuya/image_001.png)
 	
 **5.** Click **Create** to continue project configuration.
 
 **6.** In **Configuration Wizard**, select **Device status notification** API.
-   > **Note:** **Smart Home PaaS** API has been selected as default.
-   
+<p class='note'>**Smart Home PaaS** API has been selected as default.</p>
```
Comment (ludeeus) [inline source/_integrations/tuya.markdown]:
```suggestion

<p class='note'>

**Smart Home PaaS** API has been selected as default.

</p>
```

File: source/_integrations/tuya.markdown [hunk 5]
```diff
@@ -78,21 +79,25 @@ Link devices by your app account and copy the **Device ID** in the **Device List
...
-**5.** Navigate to the **Device List** tab. You can find the devices that have been added through your **Tuya Smart** or **Smart Life** account.   
+**5.** Navigate to the **All Devices** tab. You can find the devices that have been added through your **Tuya Smart** or **Smart Life** account.
+
 ![](/images/integrations/tuya/image_006.png)
 
 ### Install the Tuya Home Assistant integration
 
 #### Install the Tuya integration
 
 In **Configuration** > **Integrations** > **ADD INTEGRATION**, search for the keyword **Tuya** and select it for installation.
```
Comment (ludeeus) [inline source/_integrations/tuya.markdown]: The entire "Install the Tuya Home Assistant integration" section should be removed, it's already documented at the top of the document

**Review by ludeeus — DISMISSED** [review]

File: source/_integrations/tuya.markdown (hunk 5 above)
Comment (drthanwho) [reply source/_integrations/tuya.markdown]: What about the table with the description of the fields? Kill it or move it to the beginning?

File: source/_integrations/tuya.markdown (hunk 2 above)
Comment (drthanwho) [reply source/_integrations/tuya.markdown:49]: Added

File: source/_integrations/tuya.markdown (hunk 5 above)
Comment (ludeeus) [reply source/_integrations/tuya.markdown]: Those are fine

Comment (drthanwho) [reply source/_integrations/tuya.markdown]: Ok moved credentials table up. Made no sense to be at the bottom after removing the integration steps. Rmoved a heading there and minor other changes.

File: source/_integrations/tuya.markdown [hunk 6]
```diff
@@ -28,6 +28,20 @@ The Tuya integration allows integrating all Powered by Tuya devices you have add
 
 {% include integrations/config_flow.md %}
 
+Consult the following table and enter your Tuya credentials.
+
+| Field | Description |
```
Comment (frenck) [inline source/_integrations/tuya.markdown]: Please don't use tables for this (use `configuration_basic` blocks instead).

Additionally, this is information that should be showing in the dialogs. It is not common to add this to the integration description page.

File: source/_integrations/tuya.markdown [hunk 7]
```diff
@@ -28,6 +28,20 @@ The Tuya integration allows integrating all Powered by Tuya devices you have add
...
+| ------- | -------- |
+| Access ID and Access Secret| Go to your cloud project on [Tuya IoT Platform](https://iot.tuya.com/). Find the **Access ID** and **Access Secret** on the **Project Overview** tab.|
```
Comment (frenck) [inline source/_integrations/tuya.markdown]: This is already described below?

File: source/_integrations/tuya.markdown [hunk 8]
```diff
@@ -28,6 +28,20 @@ The Tuya integration allows integrating all Powered by Tuya devices you have add
...
+| Mobile App | Must select the one you used to link devices on the Tuya IoT Platform. |
+| Country Code | The country you select on logging in to Tuya Smart or Smart Life app.|
+| Account | Tuya Smart or Smart Life app account. |
+| Password | The password of your app account. |
+
+<p class='note'>
+The app mentioned in the table must be the one you used to link devices on the Tuya IoT Platform.
```
Comment (frenck) [inline source/_integrations/tuya.markdown]: Feel weird that this needs to be a separate note? Why not tell that directly when writing the description of the mobile app field?

File: source/_integrations/tuya.markdown [hunk 9]
```diff
@@ -36,38 +50,51 @@ The Tuya integration allows integrating all Powered by Tuya devices you have add
...
-## Quick installation and startup
-
-### Configure on the Tuya IoT Platform
+## Configuration of the Tuya IoT Platform
 
 #### Prerequisites
 
 You have created an account for [Tuya Smart or Smart Life app](https://developer.tuya.com/en/docs/iot/tuya-smart-app-smart-life-app-advantages?id=K989rqa49rluq#title-1-Download).
 
+You have created an account for the [Tuya IoT Platform](https://iot.tuya.com/).
```
Comment (frenck) [inline source/_integrations/tuya.markdown]: That is what the above line already says?

File: source/_integrations/tuya.markdown [hunk 10]
```diff
@@ -36,38 +50,51 @@ The Tuya integration allows integrating all Powered by Tuya devices you have add
...
+
 #### Create a project
 
+<p class='note'>
+If you haven't already signed up in Tuya IoT Platform, you will need to make an account first. You cannnot log in with your app's credentials here.
+</p>
+
```
Comment (frenck) [inline source/_integrations/tuya.markdown]: This is now written twice above? Why a thrid time?

File: source/_integrations/tuya.markdown [hunk 11]
```diff
@@ -36,38 +50,51 @@ The Tuya integration allows integrating all Powered by Tuya devices you have add
...
 **1.** Log in to the [Tuya IoT Platform](https://iot.tuya.com/).
 
-**2.** In the left navigation bar, click **Cloud** > **Projects**.
+**2.** In the left navigation bar, click **Cloud** > **Development**.
 
-**3.** On the page that appears, click **Create**.
+<p class='note'>
```
Comment (frenck) [inline source/_integrations/tuya.markdown]: Does this have to be a note block? Or can we just make this part of the text?

File: source/_integrations/tuya.markdown [hunk 12]
```diff
@@ -36,38 +50,51 @@ The Tuya integration allows integrating all Powered by Tuya devices you have add
...
 	
 **5.** Click **Create** to continue project configuration.
 
 **6.** In **Configuration Wizard**, select **Device status notification** API.
-   > **Note:** **Smart Home PaaS** API has been selected as default.
-   
+
+<p class='note'>
+
+**Smart Home PaaS** API has been selected as default.
```
Comment (frenck) [inline source/_integrations/tuya.markdown]: Does this have to be a big note block?

**Review by frenck — CHANGES_REQUESTED** [review]

**Comment by frenck** [comment]
Honestly, I think the order of the page is wrong. Prerequisites should be shown above the Home Assistant configuration instructions. Currently, it is the wrong way around, causing it to be a bit weird in terms of order/flow to set things up.

**Comment by drthanwho** [comment]
> Honestly, I think the order of the page is wrong. Prerequisites should be shown above the Home Assistant configuration instructions. Currently, it is the wrong way around, causing it to be a bit weird in terms of order/flow to set things up.

Yeah that is actually a very good point about the prerequisites. You won't be able to do anything without those steps. Let me see what I can do. But not just the prerequisites right? All the instructions for IoT platform first then the MyHA button.

**Comment by frenck** [comment]
>  All the instructions for IoT platform first then the MyHA button.

Yes :)

File: source/_integrations/tuya.markdown (hunk 6 above)
Comment (drthanwho) [reply source/_integrations/tuya.markdown]: Deleted

File: source/_integrations/tuya.markdown (hunk 11 above)
Comment (drthanwho) [reply source/_integrations/tuya.markdown]: Nope it does not

**Comment by drthanwho** [comment]
Ok maybe this is better now? Removed all NOTE windows in the end. I am still not a big fan of the huge troubleshooting table. But didn't touch it for now.

Oh and three new screenshot but will add after this. If order is changed should all screenshots be readded on correct order? Since now last screenshot is 003.

File: source/_integrations/tuya.markdown (hunk 9 above)
Comment (ludeeus) [reply source/_integrations/tuya.markdown]: The line above is for the app, this line is for the IoT platform, you need both (no the same account)

**Comment by frenck** [comment]
@drthanwho There is a merge conflict (due to change made in the config flow). Could you rebase this PR to resolve that conflict?

**Comment by drthanwho** [comment]
Yeah I'm gonna start all over again with a new PR. I have no idea what I'm doing with trying to rebase and clearly it's not working

**Comment by drthanwho** [comment]
Let's forget this one ever happened 😅

**Pull request closed without merging**